### DISCUSSION TITLE
URGENT: Module not found: Can't resolve 'fs'

### DISCUSSION THREAD

**Discussion by JustinBrenner** [vercel/next.js#49481] [Help]
### Summary

Hey all, 

I have tried just about everything to resolve this but now i need some help. I have had a project running for a year now untouched, i went in to update a couple of env variable path names and pushed back to github and vercel and i get the following error: 

`- error ./node_modules/next/dist/compiled/@vercel/og/index.node.js:17950:0
Module not found: Can't resolve 'fs'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/next/dist/server/web/spec-extension/image-response.js
./node_modules/next/server.js
./node_modules/@supabase/auth-helpers-nextjs/dist/index.js
./src/pages/_app.tsx`

I am running next 13.0.7 and for the life of me cannot figure out what is going on.  Like i said the project has been running untouched perfectly for a year. I have referenced everything available and tried a multitude of different solutions without success. 
![Untitled](https://user-images.githubusercontent.com/71353662/236928357-9bfe6af8-013c-4e30-ae98-53ab90aaa885.png)

**Comment by aliber4079** [comment]
It happens when server code somehow makes it to the code intended for the browser, because the browser doesn't have the fs module.

**Comment by icyJoseph** [comment]
Could you show what's going on in `_app`? Did you make any changes there? or to any of its imports? Did you update packages?

It'd seem like some package is not making use `fs` on `_app`, which would crash out when trying to make a browser build. 

It might be a transitive problem. `_app` imports `moduleA`, which uses `moduleB`, which in turn has some code that is meant for server side only. If this happens inside node_modules because a 3rd party moved code around, then it might "just happen", when trying to update a "few things" on a project.

**Comment by jkiss** [comment]
if you use some function like `getStaticProps`? if so, it's quite possible that nextjs get data from server failed at build time and try to run `getStaticProps` on the client.

**Comment by secret-username** [comment]
I just ran into the same problem. I found a solution that worked fro me here:
https://bobbyhadz.com/blog/module-not-found-cant-resolve-fs

**Reply by hallodom** [reply]
Legend thank you!
